transformer_engine/pytorch/utils.py

```diff
@@ -414,3 +424,16 @@ def canonicalize_process_group(
     if group is None:
         return torch.distributed.distributed_c10d._get_default_group()
     return group
+
+
+def torch_get_autocast_gpu_dtype() -> torch.dtype:
+    """Get PyTorch autocast GPU dtype."""
+    if is_torch_min_version("2.4.0a0"):
```

Comment:
Generally LGTM, via the `a0` suffix everywhere?

Reply:
The PyTorch version in NGC PyTorch container is tagged in a way like `2.7.0a0+7c8ec84dab`. `packaging.version.Version` is smart enough to have 
```python
In [1]: from packaging.version import Version

In [2]: Version("2.7.0a0+7c8ec84dab") >= Version("2.7a0")
Out[2]: True

In [3]: Version("2.7.0") >= Version("2.7a0")
Out[3]: True
```

Reply:
Just realize that there's already a cached PyTorch version check in TE https://github.com/NVIDIA/TransformerEngine/blob/4ceb3d4c51d09cc66e080e62c3e45f9113d75683/transformer_engine/pytorch/__init__.py#L25-L28, so I don't intend to introduce another one like `is_torch_min_version()`.

I replaced all `is_torch_min_version()` with `torch_version() >= (x, y, x)` now. And I find that this `PkgVersion(str(torch.__version__)).release` will interpret  `2.7.0a0+7c8ec84dab` as `(2, 7, 0)`, so `a0` is not needed anymore.

Reply:
Nice, I prefer using tuples for version comparisons. See https://github.com/NVIDIA/TransformerEngine/pull/1722#discussion_r2065049457.